test/e2e/Makefile

```diff
@@ -0,0 +1,138 @@
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-md-remediation > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-md-remediation.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-zone > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-zone.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-account > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-account.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-domain > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-domain.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-template > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-template.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-cp-offering > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-cp-offering.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-worker-offering > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-worker-offering.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-insufficient-compute-resources > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-insufficient-compute-resources.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-node-drain > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-node-drain.yaml
+
+## --------------------------------------
+## Docker build and push to local registry
+## --------------------------------------
+
+.PHONY: docker-build-and-push
+docker-build-and-push:
+	-d=$$(date +%s) \
+	&& IMG=$(IMG) $(MAKE) -C $(REPO_ROOT) docker-build \
+	&& IMG=$(IMG) $(MAKE) -C $(REPO_ROOT) docker-push \
+	&& echo "This step took $$(($$(date +%s)-d)) seconds"
+
+
+## --------------------------------------
+## End-to-End Tests
+## --------------------------------------
+
+.PHONY: e2e-essentials
+e2e-essentials: $(GINKGO) kind-cluster manifests cluster-templates docker-build-and-push
+
+.PHONY: run-e2e
+run-e2e: e2e-essentials ## Run End-to-End Tests
+	-d=$$(date +%s) \
+	&& PATH=$(BIN_DIR):${PATH} $(GINKGO) -v -trace -tags=e2e -skip=Conformance -nodes=1 --noColor=false ./... -- \
+	    -e2e.artifacts-folder=${REPO_ROOT}/_artifacts \
+	    -e2e.config=${REPO_ROOT}/test/e2e/config/cloudstack.yaml \
+	    -e2e.skip-resource-cleanup=false -e2e.use-existing-cluster=true \
+	&& echo "This step took $$(($$(date +%s)-d)) seconds"
+	kind delete clusters capi-test
+
+.PHONY: run-e2e-pr-blocking
```

Comment:
As far as I can tell, the only thing different between all these targets is the tag passed to ginkgo?

If so, can we just make the tag an arg the same way we run local jobs in the Prow repo?

[run job](https://github.com/aws/cluster-api-provider-cloudstack-prow-jobs/blob/cbe81fadfb11bc4431022df342e1fc11e792100a/Makefile#L47)